### ISSUE TITLE
Multiple issues with clearing more than one order at a time from the order books

### ISSUE THREAD

**Issue by ElectricDosh** [The-GNTL-Project/Exchange#84]
I'll use whole numbers and ignore fees for simplicity.

Example: there are 3 orders in the sell book to sell 1 XMR, one for 30 GNTL, one for 40 GNTL, and one for 50 GNTL.
I place a buy order at a price of 50 GNTL and commit a total of 150 GNTL.

Expected behaviour: I receive 3 XMR for 120 GNTL, and leave a buy order in the book for 0.6 XMR for the remaining 30 GNTL at a price of 50 GNTL.
Actual behaviour: I receive 3 XMR and nothing is left over.

In addition, a trade that clears more than one order from the book leaves a massive discrepancy off by several orders of magnitude in the Cost column in the user's trade history. As an example I did a trade that cleared 3 sell orders from the book for a total of ~45 GNTL and the cost shows as 111881.5949 GNTL.

**Comment by tylerjohneddy** [comment]
When you receive your 3 xmr you get your change in GNTL back too right?

**Comment by ElectricDosh** [comment]
> When you receive your 3 xmr you get your change in GNTL back too right?

Nope, as far as I can tell the full 150 is taken and nothing is returned. I'll do another test and see what happens on the seller's end.

**Comment by ElectricDosh** [comment]
Just tested and the seller does not receive the missing funds, so they're just disappearing.

**Comment by tylerjohneddy** [comment]
i think i have fixed this. ill  upload the fix soon and post when done

**Comment by BKdilse** [comment]
@tylerjohneddy this is better, the totals balance out correctly.

**1 ARQ selling for 30, 40, 50 GNTL:**
Seller receives 120 GNTL, and looses 3 ARQ+Fees
Buyer receives 3 ARQ, and looses 120 GNTL+Fees

**My Completed trades does not look correct:**
Seller's one looks fine, with 3 orders at correct prices.
Buyer's one looks wrong, with 1 order at price of 50 GNTL.

**Comment by BKdilse** [comment]
@tylerjohneddy I've Re-tested this.

Seller's balances and Completed trades look good, 3 individual trades shown.

Buyers balances look good, but Completed trades has 1 entry @ Price 50 GNTL, 3 ARQ, 120 GNTL Total.  When clicking the order buy (50, 3rd in the list), it showed spend of 150 GNTL, instead of 120 GNTL.  DB shows 1st amount of 150 GNTL and price in firstticker of 50 GNTL.

**Comment by BKdilse** [comment]
As the balances for buyer and seller are correct, this appears to be a display issue.  When clicking the order, the total is 150 GNTL, but once trade completes, it shows 120 GNTL.

**Comment by tylerjohneddy** [comment]
so this is a front end issue? and a backend issue stroing the incorrect values in the DB?

**Comment by BKdilse** [comment]
Yes looks that way.

**Comment by tylerjohneddy** [comment]
okay, are you able to look at the front end? and take a image of the issue in the DB and send it over?

**Comment by BKdilse** [comment]
Seller:  30, 40, 50
![image](https://user-images.githubusercontent.com/36295058/140802137-20b0d5fa-5b07-4197-89ee-2bd5081d4ffe.png)

Buyer clicks last order (50):
![image](https://user-images.githubusercontent.com/36295058/140802246-6e61eb24-4bf0-4abb-9027-0d8e7ddfecc8.png)

Seller completed:
![image](https://user-images.githubusercontent.com/36295058/140802389-41398d35-b966-4b6f-8637-b49a5cbeed46.png)
![image](https://user-images.githubusercontent.com/36295058/140802467-113fa3f7-c8ed-4ee6-8c8c-ad52b0cd48a8.png)

Buyer completed:
![image](https://user-images.githubusercontent.com/36295058/140802512-b7e0e811-3179-4280-bea1-6969f59ec122.png)
![image](https://user-images.githubusercontent.com/36295058/140802547-8a7aaa44-d36b-42f9-9ed6-6ca822977e2b.png)

DB:
![image](https://user-images.githubusercontent.com/36295058/140802711-ec802c30-70fa-4563-b08c-c2aba6630227.png)



Oh, the Fee wallet gained 0.30 (which is for 150 GNTL and not 120 GNTL)
